x-pack/plugins/spaces/public/share_saved_objects_to_space/components/share_to_space_flyout_internal.test.tsx

```diff
@@ -0,0 +1,745 @@
+  const onClose = jest.fn();
+  const onUpdate = jest.fn();
+
+  const mockSpacesManager = spacesManagerMock.create();
+
+  // note: this call is made in the SpacesContext
+  mockSpacesManager.getActiveSpace.mockResolvedValue({
+    id: 'my-active-space',
+    name: 'my active space',
+    disabledFeatures: [],
+  });
+
+  // note: this call is made in the SpacesContext
+  mockSpacesManager.getSpaces.mockResolvedValue(
+    opts.mockSpaces || [
+      {
+        id: 'space-1',
+        name: 'Space 1',
+        disabledFeatures: [],
+      },
+      {
+        id: 'space-2',
+        name: 'Space 2',
+        disabledFeatures: [],
+      },
+      {
+        id: 'space-3',
+        name: 'Space 3',
+        disabledFeatures: [],
+      },
+      {
+        id: 'my-active-space',
+        name: 'my active space',
+        disabledFeatures: [],
+      },
+    ]
+  );
+
+  mockSpacesManager.getShareSavedObjectPermissions.mockResolvedValue({
+    shareToAllSpaces: opts.canShareToAllSpaces ?? true,
+  });
+
+  const savedObjectToShare = {
+    type: 'dashboard',
+    id: 'my-dash',
+    namespaces: opts.namespaces || ['my-active-space', 'space-1'],
+    icon: 'dashboard',
+    title: 'foo',
+  };
+
+  const { getStartServices } = coreMock.createSetup();
+  const startServices = coreMock.createStart();
+  startServices.application.capabilities = {
+    ...startServices.application.capabilities,
+    spaces: { manage: true },
+  };
+  const mockToastNotifications = startServices.notifications.toasts;
+  getStartServices.mockResolvedValue([startServices, , ,]);
+
+  const SpacesContext = getSpacesContextWrapper({
+    getStartServices,
+    spacesManager: mockSpacesManager,
+  });
+  const ShareToSpaceFlyout = getShareToSpaceFlyoutComponent();
+  // the internal flyout depends upon the Kibana React Context, and it cannot be used without the context wrapper
+  // the context wrapper is only split into a separate component to avoid recreating the context upon every flyout state change
+  // the ShareToSpaceFlyout component renders the internal flyout inside of the context wrapper
+  const wrapper = mountWithIntl(
+    <SpacesContext feature={opts.mockFeatureId}>
+      <ShareToSpaceFlyout
+        savedObjectTarget={savedObjectToShare}
+        onUpdate={onUpdate}
+        onClose={onClose}
+        enableCreateCopyCallout={opts.enableCreateCopyCallout}
+        enableCreateNewSpaceLink={opts.enableCreateNewSpaceLink}
+        behaviorContext={opts.behaviorContext}
+      />
+    </SpacesContext>
+  );
+
+  // wait for context wrapper to rerender
+  await act(async () => {
+    await nextTick();
+    wrapper.update();
+  });
+
+  if (!opts.returnBeforeSpacesLoad) {
+    // Wait for spaces manager to complete and flyout to rerender
+    wrapper.update();
+  }
+
+  return { wrapper, onClose, mockSpacesManager, mockToastNotifications, savedObjectToShare };
+};
+
+describe('ShareToSpaceFlyout', () => {
+  beforeAll(() => {
+    jest.useFakeTimers();
```

Comment:
__nit:__ I haven't notice any tests that uses Jest fake timers functionality, do we really need to fake timers?

Reply:
Good catch, we don't need this. It's cruft that was copied over when I modeled this component after the CopyToSpaceFlyout. I'll remove it here and there.